### ISSUE TITLE
False positive for prefer_initializing_formals

### ISSUE THREAD

**Issue by a14n** [dart-lang/linter#493]
In Flutter there's 4 lints (every elements in the initialization list) for [the following code](https://github.com/flutter/flutter/blob/master/examples/layers/rendering/src/sector_layout.dart#L20-L24):

```dart
  const SectorConstraints.tight({ double deltaRadius: 0.0, double deltaTheta: 0.0 })
    : minDeltaRadius = deltaRadius,
      maxDeltaRadius = deltaRadius,
      minDeltaTheta = deltaTheta,
      maxDeltaTheta = deltaTheta;
```

**Comment by alexeieleusis** [comment]
I guess the fix consists in not linting if a parameter gets assigned to more than one field. I kind of like it, and definitely prefer it to lose the final modifier on fields.

**Comment by alexeieleusis** [comment]
Nice finding, thanks!

**Comment by alexeieleusis** [comment]
@JPaulsen had already considered this case:

```dart

class SectorConstraints {
  final double minDeltaRadius;
  final double maxDeltaRadius;
  final double minDeltaTheta;
  final double maxDeltaTheta;

  const SectorConstraints.tight(
      { this.minDeltaRadius: 0.0, this.minDeltaTheta: 0.0 })
      : maxDeltaRadius = minDeltaRadius,
        maxDeltaTheta = minDeltaTheta;
}
```
